server/sonar-server/src/test/java/org/sonar/server/projectlink/ws/SearchActionTest.java

```diff
@@ -0,0 +1,202 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.projectlink.ws;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.DbTester;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.component.ComponentLinkDto;
+import org.sonar.server.component.ComponentFinder;
+import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.WsActionTester;
+import org.sonarqube.ws.WsProjectLinks;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.core.util.Uuids.UUID_EXAMPLE_01;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.test.JsonAssert.assertJson;
+import static org.sonarqube.ws.MediaTypes.PROTOBUF;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.PARAM_PROJECT_ID;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.PARAM_PROJECT_KEY;
+
+public class SearchActionTest {
+
+  private final String PROJECT_KEY = KEY_PROJECT_EXAMPLE_001;
+  private final String PROJECT_UUID = UUID_EXAMPLE_01;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+  DbClient dbClient = db.getDbClient();
+  DbSession dbSession = db.getSession();
+
+  WsActionTester ws;
+
+  SearchAction underTest;
+
+  @Before
+  public void setUp() {
+    ComponentFinder componentFinder = new ComponentFinder(dbClient);
+    underTest = new SearchAction(dbClient, userSession, componentFinder);
+    ws = new WsActionTester(underTest);
+
+    userSession.login("login").setGlobalPermissions(SYSTEM_ADMIN);
+  }
+
+  @Test
+  public void test_example() {
+    ComponentDto project =  insertProject();
+    insertHomepageLink(project.uuid());
+    insertCustomLink(project.uuid());
+
+    String result = ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .execute().getInput();
+
+    assertJson(result).ignoreFields("id").isSimilarTo(getClass().getResource("list-example.json"));
+  }
+
+  @Test
+  public void request_by_project_id() throws IOException {
+    ComponentDto project =  insertProject();
+    insertHomepageLink(project.uuid());
+
+    InputStream responseStream = ws.newRequest()
```

Comment:
create a `callByUuid(@Nullable String uuid)` and `callByKey(@Nullable String key)` method which returns directly the SearchWsResponse object and set the MediaType